src/cli/service/echo.rs

```diff
@@ -221,8 +228,34 @@ impl<H> EchoServiceBuilder<H> {
             http_version: self.http_version,
 
             http_service_builder: (self.http_service_builder, layer),
+
+            uadb: self.uadb,
         }
     }
+
+    /// set the user agent datasbase that if set would be used to look up
+    /// a user agent (by ua header string) to see if we have a ja3/ja4 hash.
+    pub fn user_agent_database(mut self, db: std::sync::Arc<UserAgentDatabase>) -> Self {
+        self.uadb = Some(db);
+        self
+    }
+
+    /// maybe set the user agent datasbase that if set would be used to look up
+    /// a user agent (by ua header string) to see if we have a ja3/ja4 hash.
+    pub fn maybe_user_agent_database(
```

Comment:
```suggestion
    pub fn maybe_with_user_agent_database(
```

so far I've used `maybe_with` in codebase AFAIK